Sprint-1/interpret/time-format.js

```diff
@@ -13,12 +13,76 @@ console.log(result);
 
 // a) How many variable declarations are there in this program?
 
+/* Total 6 (six):
+const movieLength
+const remainingSeconds
+const totalMinutes
+const remainingMinutes
+const totalHours
+const result
+ */
+
 // b) How many function calls are there?
 
+/* Taking into account the string formatting with `$`, 
+there are 4 in total:
+line 9: $ 3(three) times
```

Comment:
Why do you consider `${...}` in the string template literal a function call? And what function was called?

Reply:
I already wrote above that theory is not my strong point. I can only answer that it is not me who considers it a function. Perhaps, say that the JS interpreter consider such as, it will be more correct. Maybe the language design, honestly I do not know how to explain it correctly. I just know that is a function, if you do a decomposition, for example, to the level of Assembler or machine code. I hope I was able to defend my opinion, assertion and understanding

If I am wrong, I will be happy to hear your more expert opinion and will take it into service

Reply:
You can always try ChatGPT for a second opinion, even though it may not always give correct answer.

And that's why I did not say you were right or wrong because I don't know exactly how the JS interpreter do when it parse the template string literal. For all I know, the interpreter can indeed be calling some function to obtain the string value  in the ${....] expression.

Reply:
Thanks to CJ for this discussion.

I tortured ChatGPT with questions and this is what I got in the end:
I understand what you mean by an "internal function" of the interpreter. Yes, ***interpolation*** can be thought of as a kind of built-in interpreter function that automatically evaluates expressions and converts them to strings. Although interpolation isn’t a standalone function, the interpreter does handle ${...}, evaluating the expression inside and inserting the result into the string. In this sense, it’s an internal mechanism, somewhat like a “hidden” function that the interpreter executes to simplify string handling.  😊